controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -703,60 +703,28 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_WSC(const std::string &sr
-    bool intel_agent                                                       = false;
-    int type;
-
-    while ((type = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
-        if (type == int(wfa_map::eTlvTypeMap::TLV_AP_RADIO_BASIC_CAPABILITIES)) {
-            radio_basic_caps = cmdu_rx.addClass<wfa_map::tlvApRadioBasicCapabilities>();
-            if (!radio_basic_caps)
-                return false;
-            LOG(DEBUG) << "Found TLV_AP_RADIO_BASIC_CAPABILITIES TLV";
-        } else if (type == int(ieee1905_1::eTlvType::TLV_WSC)) {
-            tlvWsc = cmdu_rx.addClass<ieee1905_1::tlvWsc>();
-            if (!tlvWsc)
-                return false;
-            LOG(DEBUG) << "Found TLV_WSC TLV";
-        } else if (type == int(ieee1905_1::eTlvType::TLV_VENDOR_SPECIFIC)) {
-            // If this is an Intel Agent, it will have VS TLV as the last TLV.
-            // Currently, we don't support skipping TLVs, so if we see a VS TLV, we assume
-            // It is an Intel agent, and will add the class in the Intel agent handling below.
-            intel_agent = true;
-            break;
-        } else if (type == int(wfa_map::eTlvTypeMap::TLV_AP_RADIO_IDENTIFIER)) {
-            // Check if this is a M2 message that we sent to the agent, which was just looped back.
-            // The M1 and M2 messages are both of CMDU type AP_Autoconfiguration_WSC. Thus,
-            // when we send the M2 to the local agent, it will be published back on the local bus because
-            // the destination is our AL-MAC, and the controller does listen to this CMDU.
-            // Ideally, we should use the message type attribute from the WSC payload to distinguish.
-            // Unfortunately, that is a bit complicated with the current tlv parser. However, there is another
-            // way to distinguish them: the M1 message has the AP_Radio_Basic_Capabilities TLV,
-            // while the M2 has the AP_Radio_Identifier TLV.
-            // If this is a looped back M2 CMDU, we can treat is as handled successfully.
-            LOG(DEBUG) << "Loopbed back M2 CMDU";
-            return true;
-        } else {
-            LOG(ERROR) << "Unknown TLV, type " << std::hex << type;
-            return false;
-        }
-        type = cmdu_rx.getNextTlvType();
-    }
-
-    if (!radio_basic_caps) {
-        LOG(ERROR) << "Failed to get APRadioBasicCapabilities TLV";
-        return false;
-    }
-
+    auto radio_basic_caps = cmdu_rx.getClass<wfa_map::tlvApRadioBasicCapabilities>();
+    auto tlvWsc           = cmdu_rx.getClass<ieee1905_1::tlvWsc>();
     if (!tlvWsc) {
-        LOG(ERROR) << "Failed to get TLV_WSC TLV";
+        LOG(ERROR) << "getClass<ieee1905_1::tlvWsc> failed";
         return false;
     }
-    auto m1 = std::dynamic_pointer_cast<WSC::m1>(WSC::AttrList::parse(*tlvWsc));
+    auto m1 = std::dynamic_pointer_cast<WSC::m1>(tlvWsc->getInnerClassList());
```

Comment:
Again, it would make more sense to me do do the inner parsing here instead of doing a down-cast.

Reply:
Done.